data_source.go

```diff
@@ -103,6 +108,97 @@ func (ds *AnySource) StartRun() error {
 	return nil
 }
 
+// makeDirectory creates directory of the form basepath/20060102/000 where
+// the 3-digit subdirectory counts separate file-writing occasions.
+// It also returns the formatting code for use in an Sprintf call
+// basepath/20060102/000/20060102_run000_%s.ljh and an error, if any.
+func makeDirectory(basepath string) (string, error) {
+	today := time.Now().Format("20060102")
+	todayDir := fmt.Sprintf("%s/%s", basepath, today)
+	if err := os.MkdirAll(todayDir, 0755); err != nil {
+		return "", err
+	}
+	for i := 0; i < 10000; i++ {
+		thisDir := fmt.Sprintf("%s/%4.4d", todayDir, i)
+		_, err := os.Stat(thisDir)
+		if os.IsNotExist(err) {
+			if err2 := os.MkdirAll(thisDir, 0755); err2 != nil {
+				return "", err
+			}
+			return fmt.Sprintf("%s/%s_run%4.4d_%%s.ljh", thisDir, today, i), nil
+		}
+	}
+	return "", fmt.Errorf("out of 4-digit ID numbers for today in %s", todayDir)
+}
+
+// WriteControl changes the data writing start/stop/pause/unpause state
+func (ds *AnySource) WriteControl(config *WriteControlConfig) error {
+	if strings.HasPrefix(config.Request, "Pause") {
```

Comment:
should we be case insensitive here? are we elsewhere?

Reply:
Yeah, probably should make this case-insensitive. See next push.